pkg/api/openshiftcluster.go

```diff
@@ -32,6 +35,29 @@ func (oc *OpenShiftCluster) UsesWorkloadIdentity() bool {
 	return oc.Properties.PlatformWorkloadIdentityProfile != nil && oc.Properties.ServicePrincipalProfile == nil
 }
 
+// ClusterMsiResourceId returns the resource ID of the cluster MSI or an error
+// if it encounters an issue while grabbing the resource ID from the cluster
+// doc. It is written under the assumption that there is only one cluster MSI
+// and will have to be refactored if we ever use more than one.
+func (oc *OpenShiftCluster) ClusterMsiResourceId() (*arm.ResourceID, error) {
```

Comment:
This relies on the assumption that the UserAssignedIdentities will have a single identity always. I know we are validating it as part of static validation but do you want to make sure we return the clusterMsi on the first iteration itself, or add an additional check here?

Reply:
Goutham had asked a similar question, and I responded: https://github.com/Azure/ARO-RP/pull/3757#discussion_r1707873400

But since this has come up twice, there's probably value in changing this function to explicitly check for one cluster MSI instead of using a loop. I'll make the change.

Reply:
Done!